### PR TITLE
Update contract for anomaly detector

### PR DESCRIPTION
**Pull request by guinao**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changes Made.
Add 'none' and 'microsecond' enum value into the 'granularity' field. Change 'timestamp' filed in 'series' from required to optional.

### Validation.
According to the meeting with @JeffreyRichter , @johanste , 
Tests against all the following sdks, none of them break.
| language | sdk | result |
|  :----  | :----  | :----  |
| C# | Azure.AI.Anomaly v3.0.0-preview.2 | passed |
| Python | azure-ai-anomalydetector 3.0.0b2 | passed |
| Node JS | @azure/ai-anomaly-detector 3.0.0-beta.2 | passed |
| Java | azure-ai-anomalydetector 3.0.0-beta.1 | passed |

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [x] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 1]
```diff
@@ -226,9 +226,65 @@
...
+        "secondly",
+        "microsecond"
       ]
     },
+    "ImputeMode": {
+      "type": "string",
+      "description": "can be one of auto, previous, linear, fixed, zero or notFill.",
+      "x-nullable": false,
+      "x-ms-enum": {
+        "name": "TimeGranularity",
```
Comment (jhendrixMSFT) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:238]: There's already an enum type with this name, one of them will have to be renamed.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 1 above)
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: Updated.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 2]
```diff
@@ -226,9 +226,65 @@
...
         "hourly",
         "minutely",
-        "secondly"
+        "secondly",
+        "microsecond"
       ]
     },
+    "ImputeMode": {
+      "type": "string",
+      "description": "can be one of auto, previous, linear, fixed, zero or notFill.",
```
Comment (jorgecotillo) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:235]: nit, capital "C"

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 3]
```diff
@@ -359,6 +417,15 @@
             "type": "boolean",
             "x-nullable": false
           }
+        },
+        "severity": {
+          "type": "array",
```
[the remark this answers is not in the document]
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:422]: Should other fileds also be marked as readOnly as well?

---
In reply to: [560688590](https://github.com/Azure/azure-rest-api-specs/pull/12487#discussion_r560688590) [](ancestors = 560688590)

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 2 above)
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:235]: Updated.

---
In reply to: [560686437](https://github.com/Azure/azure-rest-api-specs/pull/12487#discussion_r560686437) [](ancestors = 560686437)

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 4]
```diff
@@ -226,9 +226,65 @@
         "daily",
         "hourly",
         "minutely",
-        "secondly"
+        "secondly",
+        "microsecond"
```
Comment (jhendrixMSFT) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:230]: You also need to add `microsecond` to the `x-ms-enum` list for this enum.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jorgecotillo) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:422]: Yes, if they are set by your service, see [here](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi019-annotate-read-only-properties-with-readonly-true)

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 5]
```diff
@@ -411,6 +478,11 @@
         "isPositiveAnomaly": {
           "type": "boolean",
           "description": "Anomaly status in positive direction of the latest point. True means the latest point is an anomaly and its real value is larger than the expected one."
+        },
+        "severity": {
```
[the remark this answers is not in the document]
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:482]: Actually our service is going to GA. We just have another PR to change 'preview' to 'stable' #12472. https://github.com/Azure/azure-rest-api-specs/pull/12472

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:438]: According to the following rule, I changed the property of our output model object to readOnly and removed the 'required' property.

R2056 RequiredReadOnlyProperties
Category : SDK Error

Applies to : ARM and Data plane OpenAPI(swagger) specs

Output Message: Property '{0}' is a required property. It should not be marked as 'readonly'.

Description: A model property cannot be both readOnly and required. A readOnly property is something that the server sets when returning the model object while required is a property to be set when sending it as a part of the request body.

Why the rule is important: SDK generation fails when this rule is violated.

How to fix the violation: Ensure that the given property is either marked as readonly: true or required but not both.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 4 above)
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:233]: updated.

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (jorgecotillo) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:429]: Correct, if a customer sets the value, then it cannot be readonly (you can enforce setting the value by specifying required but nothing else), readonly: true is set on properties that your service sets an example is ProvisioningState, this value is set by your service (Running, Provisioning, Succeeded | Failed, etc.) so just add readonly: true to those properties that your service sets.

[the remark this answers is not in the document]
Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:429]: Thanks just updated the PR.

**Comment by jhendrixMSFT** [comment]
Can you please fix the prettier check CI failure?

**Comment by guinao** [comment]
> Can you please fix the prettier check CI failure?

Fixed.

**Comment by jhendrixMSFT** [comment]
@guinao looks good from SDK perspective.  Does anybody from your team need to review/approve?

**Comment by guinao** [comment]
> @guinao looks good from SDK perspective. Does anybody from your team need to review/approve?

We are requesting the review board to review this change. This PR could be merged once we get sign off from them.

**Review by JeffreyRichter — APPROVED** [review]

**Review by guinao — COMMENTED** [review]
@jhendrixMSFT  we have completed the review of the PR with review board, it is ok to merge :)

File: specification/cognitiveservices/data-plane/AnomalyDetector/readme.go.md [hunk 6]
```diff
@@ -31,5 +31,5 @@ These settings apply only when `--tag=release_1_0 --go` is specified on the comm
 Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.
 
 ``` yaml $(tag) == 'release_1_0' && $(go)
-output-folder: $(go-sdk-folder)/services/preview/cognitiveservices/v1.0/$(namespace)
+output-folder: $(go-sdk-folder)/services/stable/cognitiveservices/v1.0/$(namespace)
```
Comment (jhendrixMSFT) [inline specification/cognitiveservices/data-plane/AnomalyDetector/readme.go.md:34]: Can you please remove the `stable` directory.

Comment (guinao) [reply specification/cognitiveservices/data-plane/AnomalyDetector/readme.go.md]: updated.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**